recipe/patch_yaml/jupyter_server.yaml

```diff
@@ -0,0 +1,8 @@
+if:
+  name: jupyter_server
+  version: 1.*
+  timestamp_lt: 1693923438000
+then:
+  - replace_depends:
+      old: anyio >=3.1.0
+      new: anyio >=3.1.0,<4
```

Comment:
```suggestion
  - tighten_depends:
      name: anyio
      upper_bound: "4"
```

Reply:
This will produce a better patch most likely.

Reply:
Ooof. There docs of any of this? JSON Schema?

Reply:
I'll let it go the distance, and then apply to see the difference.

Reply:
Yes see the readme in the recipe dir. we validate all of this via pydantic

Reply:
Ah, didn't think to look there.

When using heavy `pydantic`... I can highly recommend:
- at a minimum
  - dumping out and publishing on the web somewhere with `.schema.json` (with an `$id`)
    - might just be committed and published at github pages for e.g. `https://conda-forge.github.io/conda-forge-pinning/recipe/patch.2023.09.05.schema.json`
- slightly harder (e.g. some RTD configuration)
  - use a documentation tool (like [sphinx-jsonschema](https://sphinx-jsonschema.readthedocs.io))
- ideally
  - have each document provide it's `$schema` so that text editors will provide feedback, e.g. diagnostics, complete, and coverage

Reply:
We have a schema dumped to json in the repo. If you put the right hook in a comment, vscode will give you guidance, for example. 

@jaimergp can give you more tips maybe